webui/react/src/hooks/useSettings.ts

```diff
@@ -1,359 +1,299 @@
-    return acc;
-  }, {} as Partial<T>);
+const useSettings = <T>(config: SettingsConfig<T>): UseSettingsReturn<T> => {
+  const {
+    auth: { user },
+  } = useStore();
+  const { isLoading, querySettings, state, update } = useContext(UserSettings);
+  const navigate = useNavigate();
 
-  return queryString.stringify(fullSettings);
-};
+  // creates an entry at the global state
+  useEffect(() => {
+    const settings = state.get(config.applicableRoutespace);
 
-export const getConfigKeyMap = (config: SettingsConfig): Record<RecordKey, boolean> => {
-  return config.settings.reduce((acc, prop) => {
-    acc[prop.key] = true;
-    return acc;
-  }, {} as Record<RecordKey, boolean>);
-};
+    if (settings) return;
 
-const getNewQueryPath = (
-  config: SettingsConfig,
-  basePath: string,
-  currentQuery: string,
-  newQuery: string,
-): string => {
-  // Strip out existing config settings from the current query.
-  const keyMap = getConfigKeyMap(config);
-  const params = queryString.parse(currentQuery);
-  const cleanParams = {} as Record<RecordKey, unknown>;
-  Object.keys(params).forEach((key) => {
-    if (!keyMap[key] && params[key]) cleanParams[key] = params[key];
-  });
-
-  // Add new query to the clean query.
-  const cleanQuery = queryString.stringify(cleanParams);
-  const queries = [cleanQuery, newQuery].filter((query) => !!query).join('&');
-  return `${basePath}?${queries}`;
-};
+    const newSettings: Settings = {};
 
-const defaultPathChange = {
-  querySettings: {},
-  type: PathChangeType.None,
-};
+    for (const key in config.settings) {
+      const setting = config.settings[key];
 
-const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions): SettingsHook<T> => {
-  const navigate = useNavigate();
-  const location = useLocation();
-  const storage = useStorage(options?.storagePath || config.storagePath);
-  const prevSearch = usePrevious(location.search, undefined);
-  const [settings, setSettings] = useState<T>(() => getDefaultSettings<T>(config, storage));
-  const [pathChange, setPathChange] = useState<PathChange<T>>(defaultPathChange);
-
-  const configMap = useMemo(() => {
-    return config.settings.reduce((acc, prop) => {
-      acc[prop.key] = prop;
-      return acc;
-    }, {} as Record<RecordKey, SettingsConfigProp>);
-  }, [config.settings]);
+      newSettings[setting.storageKey] = setting.defaultValue;
+    }
+
+    update(config.applicableRoutespace, newSettings);
+  }, [config, state, update]);
+
+  // parse navigation url to state
+  useEffect(() => {
+    if (!querySettings) return;
+
+    const settings = queryToSettings<T>(config, querySettings);
+    let stateSettings = state.get(config.applicableRoutespace) ?? {};
+
+    if (isEqual(settings, stateSettings)) return;
+
+    Object.keys(settings).forEach((setting) => {
+      stateSettings = Object.assign(stateSettings, { [setting]: settings[setting] });
```

Comment:
are you sure you wanna use `Object.assign` here? 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
`Object.assign` is destructive